### PR TITLE
added processSignUp controller

### PR DESCRIPTION
**Pull request by nguyntony**
I created a controllers directory that has the users directory (for now). In the users dir, I created a user file that will have the controller functions to handle signup / login actions.

The ticket was to create a hash for user passwords so I installed bcryptjs, created a salt and hash.

### CODE REVIEW

**Comment by nguyntony** [comment]
I created the login route and set up the controller fn that will handle when a user would like to login. When a user logs in they will need to provide their email (used as the unique identifier) and password. 

The fn will search for a user based on the email given, if one does not exist then the login fails. If one does exist, I will be able to retrieve their information via lookup using that email. I can now access the users password that is stored in the db. Using bcrypt.compareSync(), the first arg will be the password that the user just entered and the second arg will be the password retrieved from the db. If the two matches, it will return a boolean that will then either trigger the if/else clause. 

Sidenote: I realized that I've implemented a 'sync' form of bcrypt and I'm unsure of the differences between the sync and async versions. I know that the difference between sync and async but not sure how that translates to this usage of bcrypt!

File: backend/src/db/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  // before I can create a new user, I must first check if a current user exists with the same email.
+  // I can access my db and search for a user, if something is returned then I cannot create.
+  // I remember that working with sequelize there is a constraint that I can use to ensure uniqueness, I'm adding that to the migrations file but I have not ran the migrate.
+
+  const findUser = await User.findOne({
+    where: {
+      email,
+    },
+  });
+
```
Comment (coffee-intocode) [inline backend/src/db/controllers/user.js:18]: What happens when a user tries to create an account with an email that already exists in our database?

File: backend/src/db/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  if (!findUser) {
+    // if a user is not found, meaning the email does not exist then it will return null, so if I want this if statement to run, I can use !
+    const newUser = await User.create({
+      name,
+      email,
+      password: hash,
```
Comment (coffee-intocode) [inline backend/src/db/controllers/user.js:24]: You could also do an inline hash `let user = await db.User.create({
        name,
        email,
        password: await bcrypt.hash(password, 10),
      });`

File: backend/src/db/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      const accessToken = generateAccessToken(user);
+      // const refreshToken = jwt.sign(user, process.env.REFRESH_TOKEN_SECRET);
+      res.json({
+        message: "successful login",
+        accessToken,
+        refreshToken,
+      });
+    } else {
+      res.json({
+        message: "unsuccessful login",
```
Comment (coffee-intocode) [inline backend/src/db/controllers/user.js:60]: I would use a try catch block here so if there is an error you can respond with unsuccessful login

File: backend/src/db/controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      });
+    }
+  } else {
+    res.json({
```
Comment (coffee-intocode) [inline backend/src/db/controllers/user.js:64]: I don't think this second else will ever be called.

**Pull request merged**